File: ChatGPT/python/output/output_18a.py
# ChatGPT-generated solution will go here.
from collections import deque

def main():
    with open("input.txt") as f:
        lines = [line.strip() for line in f if line.strip()]

    # Read the first 1024 byte coordinates
    blocked = set()
    for line in lines[:1024]:
        x, y = map(int, line.split(','))
        blocked.add((x, y))

    max_coord = 70
    start = (0, 0)
    end = (max_coord, max_coord)

    # BFS to find shortest path
    queue = deque([(start[0], start[1], 0)])  # x, y, steps
    visited = set()
    visited.add(start)

    directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]

    while queue:
        x, y, steps = queue.popleft()
        if (x, y) == end:
            print(steps)
            return
        for dx, dy in directions:
            nx, ny = x + dx, y + dy
            if (
                0 <= nx <= max_coord and
                0 <= ny <= max_coord and
                (nx, ny) not in blocked and
                (nx, ny) not in visited
            ):
                visited.add((nx, ny))
                queue.append((nx, ny, steps + 1))

    print("No path found")

if __name__ == "__main__":
    main()

File: ChatGPT/clojure/output/output_18a.clj
;; ChatGPT-generated solution will go here.
(ns output_18a
  (:require [clojure.string :as str]
            [clojure.set :as set]
            [clojure.data.priority-map :refer [priority-map]]))

(defn parse-coord [line]
  (let [[x y] (str/split line #",")]
    [(Integer/parseInt x) (Integer/parseInt y)]))

(defn parse-input [lines]
  (->> lines
       (map str/trim)
       (filter seq)
       (map parse-coord)
       (take 1024)
       set))

(defn neighbors [[x y]]
  [[(inc x) y] [(dec x) y] [x (inc y)] [x (dec y)]])

(defn in-bounds? [[x y]]
  (and (<= 0 x 70) (<= 0 y 70)))

(defn bfs [start goal blocked]
  (loop [q (priority-map start 0)
         dist {start 0}]
    (if-let [[[pos cost] & rest] (seq q)]
      (if (= pos goal)
        cost
        (let [next (for [n (neighbors pos)
                         :when (and (in-bounds? n)
                                    (not (blocked n))
                                    (or (nil? (dist n))
                                        (< (inc cost) (dist n))))]
                     n)
              q' (reduce (fn [q n] (assoc q n (inc cost))) (dissoc q pos) next)
              d' (reduce (fn [d n] (assoc d n (inc cost))) dist next)]
          (recur q' d')))
      nil)))

(defn part [lines]
  (let [blocked (parse-input lines)]
    (bfs [0 0] [70 70] blocked)))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
